java/clients/src/test/java/sleeper/clients/status/report/compaction/job/CompactionJobStatusReporterTestBase.java

```diff
@@ -71,36 +72,50 @@ protected static List<CompactionJobStatus> mixedJobStatuses() {
 
         CompactionJobStatus status1 = jobCreated(job1, creationTime1);
         CompactionJobStatus status2 = jobCreated(job2, creationTime2,
-                startedCompactionRun(StatusReporterTestHelper.task(1), startedTime2));
+                startedCompactionRun(task(1), startedTime2));
         CompactionJobStatus status3 = jobCreated(job3, creationTime3,
-                finishedCompactionRun(StatusReporterTestHelper.task(1),
+                finishedCompactionRun(task(1),
                         summary(startedTime3, Duration.ofMinutes(1), 600, 300)));
         CompactionJobStatus status4 = jobCreated(job4, creationTime4,
-                failedCompactionRun(StatusReporterTestHelper.task(1),
+                failedCompactionRun(task(1),
                         new ProcessRunTime(startedTime4, Duration.ofMinutes(1)),
                         List.of("Something went wrong", "More details")));
         return Arrays.asList(status4, status3, status2, status1);
     }
 
     protected static List<CompactionJobStatus> mixedUnfinishedJobStatuses() {
         return mixedJobStatuses().stream()
-                .filter(job -> !job.isFinished())
+                .filter(CompactionJobStatus::isUnstartedOrInProgress)
                 .collect(Collectors.toList());
     }
 
-    protected static List<CompactionJobStatus> jobWithMultipleRuns() {
+    protected static List<CompactionJobStatus> jobsWithMultipleRuns() {
         CompactionJobTestDataHelper dataHelper = new CompactionJobTestDataHelper();
 
-        CompactionJob job = dataHelper.singleFileCompaction();
-        CompactionJobStatus status = jobCreated(job, Instant.parse("2022-10-12T10:00:00.001Z"),
-                startedCompactionRun(StatusReporterTestHelper.task(1), Instant.parse("2022-10-12T10:02:00.001Z")),
-                finishedCompactionRun(StatusReporterTestHelper.task(2), summary(
-                        Instant.parse("2022-10-12T10:01:15.001Z"),
-                        Duration.ofSeconds(30), 300L, 200L)),
-                finishedCompactionRun(StatusReporterTestHelper.task(1), summary(
-                        Instant.parse("2022-10-12T10:01:00.001Z"),
-                        Duration.ofSeconds(20), 300L, 200L)));
-        return Collections.singletonList(status);
+        CompactionJobStatus succeededThenFailed = jobCreated(dataHelper.singleFileCompaction(),
+                Instant.parse("2022-10-10T10:00:00.001Z"),
+                failedCompactionRun(task(2), new ProcessRunTime(
+                        Instant.parse("2022-10-10T10:01:15.001Z"), Duration.ofSeconds(30)),
+                        List.of("Compaction has already been committed", "Failed database update")),
+                finishedCompactionRun(task(1), summary(
+                        Instant.parse("2022-10-10T10:01:00.001Z"), Duration.ofSeconds(20), 200L, 100L)));
```

Comment:
Could we swap the order of these calls to make them match the variable name? It looks like they're defined backwards at the moment so you have to look at the update time to determine the order. Same for the variable `failedThenInProgress`

Reply:
Not with this test helper, it needs the runs in order latest first. It might be better to build the status object from the update records as they'd be held in the status store. If we did that they'd be sorted as part of building the status object. I'll see how widely this test helper is used and if I can get rid of it.